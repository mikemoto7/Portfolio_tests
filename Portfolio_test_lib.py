
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







