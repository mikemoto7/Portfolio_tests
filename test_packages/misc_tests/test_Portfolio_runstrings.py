import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






