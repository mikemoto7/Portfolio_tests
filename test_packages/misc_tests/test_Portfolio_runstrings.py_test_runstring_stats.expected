
Processing: user_runstring = ./Portfolio.py --stats --pfile demo_data_file.xml


Processing: --stats 
<Stats>
<general>
Total number investments = 49
Total amount = "$85,906,8.80", 100.0%
</general>
<retirement_classes>
SEP_IRA, 3, "$1,598,6.00", 1.86%
Traditional_IRA, 20, "$29,334,4.10", 34.15%
Non_Retirement, 20, "$34,092,1.30", 39.68%
Traditional 401K, 1, "$3,974,8.00", 4.63%
Rollover_IRA, 2, "$8,726,1.30", 10.16%
Roth_IRA, 3, "$8,180,8.10", 9.52%
Total number retirement_classes assets = 49
Total number retirement_classes = 6
Total amount non-retirement = "$34,092,1.30", 39.68%
Total amount retirement = "$51,814,7.50", 60.32%
Total amount = "$85,906,8.80", 100.0%
</retirement_classes>
<asset_classes>
Energy, 4, "$9,783,7.40", 11.39%
Global, 6, "$5,763,2.60", 6.71%
Healthcare, 2, "$2,564,0.10", 2.98%
Individual_Stocks, 1, "$2,587,8.60", 3.01%
Large_Blend, 3, "$4,350,8.60", 5.06%
Large_Growth, 4, "$10,724,5.50", 12.48%
Large_Value, 3, "$4,752,6.30", 5.53%
Mid_Blend_Aggress, 3, "$7,397,3.90", 8.61%
Mid_Growth, 5, "$8,548,2.90", 9.95%
Mid_Value, 5, "$5,408,2.80", 6.3%
REIT, 5, "$11,425,2.00", 13.3%
Small_Blend, 2, "$2,349,6.80", 2.74%
Small_Growth, 4, "$8,250,4.50", 9.6%
Small_Value, 2, "$2,000,6.80", 2.33%
Total percent = 99.98999999999998%
Total number asset_classes investments = 49
Total number asset_classes = 14
Total amount = $85,906,8.80
</asset_classes>
<retirement_asset_classes>
Energy:Non_Retirement, 1, "$1,989,8.00", 2.32%
Energy:Retirement, 3, "$7,793,9.40", 9.07%
Global:Non_Retirement, 2, "$893,8.30", 1.04%
Global:Retirement, 4, "$4,869,4.30", 5.67%
Healthcare:Non_Retirement, 1, "$2,150,4.60", 2.5%
Healthcare:Retirement, 1, "$413,5.50", 0.48%
Individual_Stocks:Non_Retirement, 1, "$2,587,8.60", 3.01%
Large_Blend:Non_Retirement, 1, "$2,036,3.00", 2.37%
Large_Blend:Retirement, 2, "$2,314,5.60", 2.69%
Large_Growth:Non_Retirement, 2, "$2,984,1.50", 3.47%
Large_Growth:Retirement, 2, "$7,740,4.00", 9.01%
Large_Value:Non_Retirement, 1, "$1,462,8.60", 1.7%
Large_Value:Retirement, 2, "$3,289,7.70", 3.83%
Mid_Blend_Aggress:Non_Retirement, 3, "$7,397,3.90", 8.61%
Mid_Growth:Non_Retirement, 2, "$1,644,4.90", 1.91%
Mid_Growth:Retirement, 3, "$6,903,8.00", 8.04%
Mid_Value:Non_Retirement, 2, "$2,370,4.60", 2.76%
Mid_Value:Retirement, 3, "$3,037,8.20", 3.54%
REIT:Non_Retirement, 1, "$3,444,6.20", 4.01%
REIT:Retirement, 4, "$7,980,5.80", 9.29%
Small_Blend:Non_Retirement, 1, "$1,901,9.60", 2.21%
Small_Blend:Retirement, 1, "$447,7.20", 0.52%
Small_Growth:Non_Retirement, 1, "$1,524,2.70", 1.77%
Small_Growth:Retirement, 3, "$6,726,1.80", 7.83%
Small_Value:Non_Retirement, 1, "$1,703,6.80", 1.98%
Small_Value:Retirement, 1, "$297,0.00", 0.35%
Total percent retirement = 99.98%
Total amount all retirement classes = $85,906,8.80
Total number retirement_asset_classes investments = 49
Total number existing retirement_asset_classes = 26
Total number missing retirement_asset_classes = 2
</retirement_asset_classes>
<totals_percents_table>

-----------------------------------------------------------------------------------------------------------
                          Tot        Tot    Tot        Ret        Ret    Ret    Non_Ret    Non_Ret  Non_Ret
Asset_Class         inv_count        amt    pct  inv_count        amt    pct  inv_count        amt      pct
Energy                      4   97837.40  11.39          3   77939.40  15.04          1   19898.00     5.84
Global                      6   57632.60   6.71          4   48694.30   9.40          2    8938.30     2.62
Healthcare                  2   25640.10   2.98          1    4135.50   0.80          1   21504.60     6.31
Individual_Stocks           1   25878.60   3.01          0       0.00   0.00          1   25878.60     7.59
Large_Blend                 3   43508.60   5.06          2   23145.60   4.47          1   20363.00     5.97
Large_Growth                4  107245.50  12.48          2   77404.00  14.94          2   29841.50     8.75
Large_Value                 3   47526.30   5.53          2   32897.70   6.35          1   14628.60     4.29
Mid_Blend_Aggress           3   73973.90   8.61          0       0.00   0.00          3   73973.90    21.70
Mid_Growth                  5   85482.90   9.95          3   69038.00  13.32          2   16444.90     4.82
Mid_Value                   5   54082.80   6.30          3   30378.20   5.86          2   23704.60     6.95
REIT                        5  114252.00  13.30          4   79805.80  15.40          1   34446.20    10.10
Small_Blend                 2   23496.80   2.74          1    4477.20   0.86          1   19019.60     5.58
Small_Growth                4   82504.50   9.60          3   67261.80  12.98          1   15242.70     4.47
Small_Value                 2   20006.80   2.33          1    2970.00   0.57          1   17036.80     5.00
Totals                     49  859068.80  99.99         29  518147.50  99.99         20  340921.30    99.99
                                                                                                           
0
[['Energy', '4', '97837.40', '11.39', '3', '77939.40', '15.04', '1', '19898.00', '5.84'], ['Global', '6', '57632.60', '6.71', '4', '48694.30', '9.40', '2', '8938.30', '2.62'], ['Healthcare', '2', '25640.10', '2.98', '1', '4135.50', '0.80', '1', '21504.60', '6.31'], ['Individual_Stocks', '1', '25878.60', '3.01', '0', '0.00', '0.00', '1', '25878.60', '7.59'], ['Large_Blend', '3', '43508.60', '5.06', '2', '23145.60', '4.47', '1', '20363.00', '5.97'], ['Large_Growth', '4', '107245.50', '12.48', '2', '77404.00', '14.94', '2', '29841.50', '8.75'], ['Large_Value', '3', '47526.30', '5.53', '2', '32897.70', '6.35', '1', '14628.60', '4.29'], ['Mid_Blend_Aggress', '3', '73973.90', '8.61', '0', '0.00', '0.00', '3', '73973.90', '21.70'], ['Mid_Growth', '5', '85482.90', '9.95', '3', '69038.00', '13.32', '2', '16444.90', '4.82'], ['Mid_Value', '5', '54082.80', '6.30', '3', '30378.20', '5.86', '2', '23704.60', '6.95'], ['REIT', '5', '114252.00', '13.30', '4', '79805.80', '15.40', '1', '34446.20', '10.10'], ['Small_Blend', '2', '23496.80', '2.74', '1', '4477.20', '0.86', '1', '19019.60', '5.58'], ['Small_Growth', '4', '82504.50', '9.60', '3', '67261.80', '12.98', '1', '15242.70', '4.47'], ['Small_Value', '2', '20006.80', '2.33', '1', '2970.00', '0.57', '1', '17036.80', '5.00']]
</totals_percents_table>
</Stats>

